crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -557,18 +584,29 @@ mod tests {
         let ta = Type::Any;
         let t1 = Type::IntLiteral(1);
         let t2 = KnownClass::Int.to_instance(&db);
+        // i0 = Any & ~Literal[1]
         let i0 = IntersectionBuilder::new(&db)
             .add_positive(ta)
             .add_negative(t1)
             .build();
+        // intersection = int & ~(Any & ~Literal[1])
+        // -> int & (~Any  | Literal[1])
+        // (~Any is treated as Any for adding purposes)
+        // -> (int & Any) | (int & Literal[1])
+        // -> (int & Any) | Literal[1]
         let intersection = IntersectionBuilder::new(&db)
```

Comment:
Confusing to name this intersection when it is actually a union :)

```suggestion
        let union = IntersectionBuilder::new(&db)
```

Reply:
FWIW another option here is to just assert on the displayed form of the type, `build_negative_union_de_morgan` does that

Reply:
At first wasn't a fan because of the display issues, but given it's DNF (I belive) it's unambiguous... will do that